redis/src/cluster_client.rs

```diff
@@ -79,6 +81,8 @@ impl From<BuilderParams> for ClusterParams {
             read_from_replicas: value.read_from_replicas,
             tls: value.tls,
             retry_params: value.retries_configuration,
+            connection_timeout: value.connection_timeout.unwrap_or(Duration::MAX),
```

Comment:
I think we should set a sane default of 1 second or less for `connection_timeout`.